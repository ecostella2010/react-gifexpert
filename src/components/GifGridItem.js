import React from 'react'
import PropType from 'prop-types';

export const GifGridItem = ( { title, url } ) => {

    //console.log ({id, title, url});

    return (
        <div className = "card animate__animated animate__bounce animate__fadeIn">
            <img src={ url } alt = { title }></img>
            <p> { title } </p>
            {/* { title } */}
        </div>
    )
}

GifGridItem.propType = {
    title: PropType.string.isRequired,
    url: PropType.string.isRequired
                 
}