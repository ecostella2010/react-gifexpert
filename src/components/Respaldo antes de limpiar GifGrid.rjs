//import React, { useState, useEffect } from 'react'
import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ( { category } ) => {

    //const [count, setCount] = useState(0);
    //Casi como el component dispose para inicializar

    // const [images, setImages] = useState([]);

    // const state = useFetchGifs();
    // console.log( state );
    //Al desestructurar
    //const { data, loading } = useFetchGifs( category );

    //Si queremos desestructurar renombrando
    const { data:images, loading } = useFetchGifs( category );

    // console.log( loading );
    // console.log( data );

    //Se lleva a custome hook useFetchGifs
    // useEffect ( () => {
    //     //getGifs ()
    //     getGifs( category )
    //     // .then(imgs => setImages( imgs ))
    //     .then( setImages )
    // }, [ category ]);

    // const getGifs = async() => {
    //     //console.log('category:', category);
    //     const url = `https://api.giphy.com/v1/gifs/search?q=${ encodeURI(category) }&limit=10&api_key=HQYcz1cE4MrSnWH5veByJYYTwqBXu7zz`;
    //     //console.log (url);
    //     const resp = await fetch ( url );
    //     //const data = await resp.json();
    //     //Uso RestructuraciÃ³n 
    //     const { data } = await resp.json();
    //     //console.log( data );
    //     const gifs = data.map ( img => {
    //         return {
    //             id : img.id,
    //             title : img.title,
    //             url : img.images?.downsized_medium.url
    //         }
    //     });

    //     //console.log( gifs );
    //     setImages(gifs);
    //     //console.log( images );
        
    // }

    return (
        <>
            <h3> Filtro : { category } </h3>

            {/* { loading ? 'Cargando...' : 'Data cargada' } */}

            {/* { loading ? <p> Loading... </p> : null } */}
            { loading && <p> Loading... </p> }

            <div className="card-grid">
                {
                    images.map ( img  => (
                            <GifGridItem
                                key = { img.id } 
                                // img = { img } 
                                //Para enviar todas las propiedades como propiedades independientes 
                                {...img}
                                
                            />
                    ))
                }

            </div>
        </>

        // <div>
        //     <h3> { category } </h3>
        //     <ol>
        //         {
        //             // images.map ( img => (
        //             //      <li key = { img.id } > { img.title }  </li>    
        //             // ) )

        //             //Desestructurando
        //             images.map ( ({ id, title }) => (
        //                 <li key = { id } > { title }  </li>    
        //            ) )

        //         }
        //     </ol>

        //     {/* <h3> { count } </h3>
        //     <button onClick= { () => setCount( count + 1) } > </button> */}
        // </div>
    )
}