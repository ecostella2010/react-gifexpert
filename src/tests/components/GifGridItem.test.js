import { shallow } from 'enzyme';
import React from 'react'; 
import '@testing-library/jest-dom';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en GifGridItem', () => {
    let wrapper = shallow ( < GifGridItem />);

    const title = 'Un titulo'; 
    const url = 'https://localhost/algo.jpg';

    beforeEach ( () => {
        wrapper = shallow ( < GifGridItem title = {title} url = {url} /> );
    });

    test('debe mostrar  < GifGridItem /> correstamente', () => {
        expect (wrapper).toMatchSnapshot();
    });

    test('debe de tener un pÃ¡rrafo con el title', () => {

        const p = wrapper.find('p');
        expect (p.text().trim() ).toBe ( title );
        
    });

    test('debe de tener la imagen igual al url y alt de los props', () => {
        
        const img = wrapper.find('img');

        // console.log (img.html());
        // console.log (img.props());
        // console.log (img.prop('src'));
        // console.log (img.props().src);
        // console.log (img.props().alt);

        expect(img.prop('src')).toBe (url);
        expect(img.prop('alt')).toBe (title);
    });
    
    test('debe de tener animate__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        //console.log (div.prop('className'));
        expect(className).toContain ('animate__fadeIn');
        expect(className.includes('animate__fadeIn')).toBe( true );
    });
    
    

})


