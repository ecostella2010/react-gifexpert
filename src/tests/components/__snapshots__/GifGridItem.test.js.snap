// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe mostrar  < GifGridItem /> correstamente 1`] = `
<div
  className="card animate__animated animate__bounce animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Un titulo
     
  </p>
</div>
`;
